src/helpers/cache.ts

```diff
@@ -1,23 +1,18 @@
-import { Mapping, MappingById } from '../types/general'
+import { Mapping, MappingById, AccessToken, TokenById } from '../types/general'
 
 const createIdCache = (records: Mapping[]): MappingById => {
-  return records.reduce(
-    (obj, item) => ({
-      ...obj,
-      [item.id]: item
-    }),
-    {}
-  )
+  return records.reduce((obj, item) => ({ ...obj, [item.id]: item }), {})
 }
 
 const createDomainCache = (records: Mapping[]): MappingById => {
   return records.reduce(
-    (obj, item) => ({
-      ...obj,
-      [item.fullDomain]: item
-    }),
+    (obj, item) => ({ ...obj, [item.fullDomain]: item }),
     {}
   )
 }
 
-export { createDomainCache, createIdCache }
+const createTokenCache = (records: AccessToken[]): TokenById => {
+  return records.reduce((obj, item) => ({ ...obj, [item.id]: item }), {})
+}
```

Comment:
`createTokenCache`'s implementation is the same as `createIdCache`, the only difference is their types, where `createTokenCache` takes in an array of type `AccessToken` and `createIdCache` takes in an array of type `Mapping`.

So there's a little bit of repetition which is fine because it's easy to understand what each function does and where it will be used and it's not a big deal.

I think this repetition is fine, but if we wanted to remove this repetition, we could make a generic function that does the same thing and can be applied in both situations in the same way. We could probably call it something like `mapById` since the function maps each record by the record's id property. This is how we would define this function.
```
const mapById = <T extends { id: string }>(records: T[]): { [id: string]: T | undefined } => {
  return records.reduce((obj, item) => ({ ...obj, [item.id]: item }), {})
}
```
This expression between `<` and `>` is a generic type parameter, type parameters are kind of like function parameters except they're for types, `<T>` means the type `T` can be anything, `<T extends { id: string }>` means the type `T` can be anything that has a property `id` which is type `string`, so `AccessToken` and `Mapping` would both work because they both have an `id` property that is type `string`, so we would be able to call `mapById` on an array of `AccessToken`s or an array of `Mapping`s.

This generic function is probably harder to understand so maybe just keep these functions the way they are, I just thought I would share this information because I think it's cool.